### PR TITLE
Develop deprecate broadcasting

### PR DESCRIPTION
**Pull request by vincentpierre**
[Associated Design Document](https://docs.google.com/document/d/1q_eJ5KHNcVXoEBkGxTm-3Kt4zl4_aAoeuccrdIoEfkI/edit#)

Builds on top of #2659 

In this PR : 
 - Removal of the `Control` checkbox in the BroadcastHub of the Academy
 - By default, all learning brains will be trained by Python
 - The BroadcastHub is now a list of **LearningBrains** and not **Brains**
 - In Editor : 
    - If no brains are in the Hub OR if there is no Python process to communicate with Inference
    - If there is at least one Brain in the Hub and UnityEnvironment was created without an executable : Training
 - In Executable : 
    - If launched with a UnityEnvironment : Training
    - If opened manually : Inference

TODO : 
 - Documentation

### CODE REVIEW

File: UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ public static BrainParametersProto ToProto(this BrainParameters bp, string name,
             {
                 VectorObservationSize = bp.vectorObservationSize,
                 NumStackedVectorObservations = bp.numStackedVectorObservations,
-                VectorActionSize = {bp.vectorActionSize},
+                VectorActionSize = { bp.vectorActionSize },
```
Comment (surfnerd) [inline UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs]: formatting

File: UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs [hunk 2]
```diff
@@ -126,7 +125,8 @@ public IEnumerable<string> GetModelFailedChecks()
         /// <inheritdoc />
         protected override void DecideAction()
         {
-            if (m_IsControlled)
+            m_Batcher?.SendBrainInfo(name, m_AgentInfos);
+            if (m_Batcher != null)
```
Comment (surfnerd) [inline UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs]: does a check for null need to be here?  What's the case when DecideAction is called when the batcher is not yet set?

Comment (vincentpierre) [reply UnitySDK/Assets/ML-Agents/Scripts/LearningBrain.cs]: If there is no batcher, I suppose the Inference behavior is used

**Comment by vincentpierre** [comment]
Unable to change the target branch due to a bug on Githib. Opening a separate PR targeting develop : #2669

File: UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs (hunk 1 above)
Comment (surfnerd) [reply UnitySDK/Assets/ML-Agents/Scripts/Grpc/GrpcExtensions.cs]: weird, different runs of the formatter gave me different results.

**Pull request closed without merging**